asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
Well, okay. It's been a while and I have no better ideas than `Doer interface`, so let's go with that for now. If really good ideas come in the future, we can probably still change things.